### PR TITLE
Add tests for HashUtil

### PR DESCRIPTION
**Pull request by davidmason**
Was difficult to predict the behaviour of HashUtil, so I added some unit tests for reference.

### CODE REVIEW

File: zanata-common-util/src/test/java/org/zanata/util/HashUtilTest.java [hunk 1]
```diff
@@ -61,4 +65,90 @@ public void fileUtilHashSameAsDigestWriter() throws Exception {
...
+    }
+
+    @Test
+    public void sourceHashHashesJoinedStrings() {
+        List<String> contents = new ArrayList<>();
+        contents.add("content 1");
+        contents.add("content 2");
+        contents.add("content 3");
+
+        String PIPE_JOINED = "content 1|content 2|content 3";
```
Comment (djansen-redhat) [inline zanata-common-util/src/test/java/org/zanata/util/HashUtilTest.java]: Possibly
```
String PIPE_JOINED = "content 1|content 2|content 3";
List<String> contents = Arrays.asList(PIPE_JOINED.split('|');
```
(Not that I care that much...)

File: zanata-common-util/src/test/java/org/zanata/util/HashUtilTest.java [hunk 2]
```diff
@@ -61,4 +65,90 @@ public void fileUtilHashSameAsDigestWriter() throws Exception {
...
+        List<String> emptyContents = new ArrayList<>();
+        assertThat(HashUtil.sourceHash(emptyContents), nullValue());
+    }
+
+    @Test
+    public void sourceHashWrapsEmptyInPipes() {
+        List<String> contents = new ArrayList<>();
+        contents.add("content 1");
+        contents.add("");
+        contents.add("");
```
Comment (djansen-redhat) [inline zanata-common-util/src/test/java/org/zanata/util/HashUtilTest.java]: Same comment as above, but with split('|', -1)

File: zanata-common-util/src/test/java/org/zanata/util/HashUtilTest.java [hunk 3]
```diff
@@ -61,4 +65,90 @@ public void fileUtilHashSameAsDigestWriter() throws Exception {
...
+        contents.add("content 1");
+        contents.add("content 2");
+        contents.add("content 3");
+        contents.add("content 4");
+        contents.add("content 5");
+        contents.add("content 6");
+        contents.add("content 7");
+        contents.add("content 8");
+        contents.add("content 9");
+        contents.add("content 10");
```
Comment (djansen-redhat) [inline zanata-common-util/src/test/java/org/zanata/util/HashUtilTest.java]: Same comment as above.

**Comment by djansen-redhat** [comment]
Minor comments, otherwise :+1:

**Comment by davidmason** [comment]
@djansen-redhat I prefer to leave the contents adding manual - takes a few more lines but makes it more immediately obvious what is being passed in.

**Pull request merged**